src/main/java/ca/on/oicr/pineryreports/reports/impl/BisqueProjectsStatusReport.java

```diff
@@ -0,0 +1,770 @@
+  private static final String COUNT_CATEGORY_RNA_STOCK = "RNA Stock";
+  private static final String COUNT_CATEGORY_DNA_ALIQUOT = "DNA Aliquot";
+  private static final String COUNT_CATEGORY_RNA_ALIQUOT = "RNA Aliquot";
+  private static final String COUNT_CATEGORY_WG_LIB = "WG Library";
+  private static final String COUNT_CATEGORY_EX_LIB = "EX Library";
+  private static final String COUNT_CATEGORY_TS_LIB = "TS Library";
+  private static final String COUNT_CATEGORY_DNA_10X_LIB = "DNA 10X Library";
+  private static final String COUNT_CATEGORY_DNA_LIB = "DNA Library;";
+  private static final String COUNT_CATEGORY_NN_LIB = "NN Library";
+  private static final String COUNT_CATEGORY_RNA_LIB = "RNA Library";
+  private static final String COUNT_CATEGORY_RNA_10X_LIB = "RNA 10X Library";
+  private static final String COUNT_CATEGORY_WG_SEQD = "WG Lib Seqd";
+  private static final String COUNT_CATEGORY_EX_SEQD = "EX Lib Seqd";
+  private static final String COUNT_CATEGORY_TS_SEQD = "TS Lib Seqd";
+  private static final String COUNT_CATEGORY_DNA_10X_SEQD = "DNA 10X Lib Seqd";
+  private static final String COUNT_CATEGORY_DNA_SEQD = "DNA Lib Seqd";
+  private static final String COUNT_CATEGORY_NN_SEQD = "NN Lib Seqd";
+  private static final String COUNT_CATEGORY_RNA_SEQD = "RNA Lib Seqd";
+  private static final String COUNT_CATEGORY_RNA_10X_SEQD = "RNA 10X Lib Seqd";
+
+  private static final String COUNT_CATEGORY_NON_ILL_LIB = "Non-Illumina Library";
+  private static final String COUNT_CATEGORY_NON_ILL_SEQD = "Non-Illumina Seqd";
+
+  private static final String P = "P";
+  private static final String R = "R";
+  private static final String O = "O";
+  private static final String X = "X";
+  private static final String M = "M";
+  private static final String L = "L";
+
+  private Set<String> projects = Sets.newHashSet();
+
+  List<String> ldO = Arrays.asList(LIBRARY_DESIGN_CH, LIBRARY_DESIGN_BS, LIBRARY_DESIGN_AS);
+
+  protected int columnCount = 0;
+
+  private static final Option OPT_PROJECT = CommonOptions.project(true);
+  public static final String REPORT_NAME = "projects-status";
+
+  private List<Map.Entry<String, Map<String, List<Count>>>> countsByProjectAsList; // String project, List<Count> all counts
+
+  @Override
+  public String getReportName() {
+    return REPORT_NAME;
+  }
+
+  @Override
+  public Collection<Option> getOptions() {
+    return Sets.newHashSet(OPT_PROJECT);
+  }
+
+  @Override
+  public void processOptions(CommandLine cmd) throws ParseException {
+    List<String> projx = Arrays.asList(cmd.getOptionValue(OPT_PROJECT.getLongOpt()).split(","));
+    this.projects = projx.stream().map(proj -> proj.trim()).collect(Collectors.toSet());
+  }
+
+  @Override
+  public String getTitle() {
+    return "Projects Status Report generated " + new SimpleDateFormat(DATE_FORMAT).format(new Date());
+  }
+
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException, IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 59 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)